resolwe_bio/processes/support_processors/prepare_geo.yml

```diff
@@ -0,0 +1,233 @@
+  type: data:other:geo:rnaseq
+  category: analyses
+  persistence: CACHED
+  description: >
+    Prepare RNA-Seq data for GEO upload.
+  input:
+    - name: reads
+      label: Reads
+      type: list:data:reads:fastq
+      description: >
+        List of reads objects. Fastq files will be used.
+    - name: expressions
+      label: Expressions
+      type: list:data:expression
+      description: >
+        Cuffnorm data object. Expression table will be used.
+    - name: name
+      label: Collection name
+      type: basic:string
+  output:
+    - name: tarball
+      label: GEO folder
+      type: basic:file
+    - name: table
+      label: Annotation table
+      type: basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      NAME={{ name }}_RNA-Seq
+      mkdir ${NAME}
+
+      declare -A SAMPLE_TO_PAIRED
+      
+      echo -e "File_name\tSample_name\tmd5sum\tSE/PE\tRead_length" > "${NAME}".txt
+
+      {% for read in reads %}
+        SAMPLE_NAME=`basename {{ read|sample_name }} .fastq.gz`
+        PAIRED=SE
+        MATE1_LABEL=""
+        MATE2_LABEL=""
+        {% if read|type|subtype('data:reads:fastq:paired:') %}
+          PAIRED=PE
+          MATE1_LABEL=_R1
+          MATE2_LABEL=_R2
+        {% endif %}
+        SAMPLE_TO_PAIRED["${SAMPLE_NAME}"]=$PAIRED
+
+        {% for fastq in read.fastq %}
+          cp {{ fastq.file }} "${NAME}/${SAMPLE_NAME}${MATE1_LABEL}.fastq.gz"
+          MD5=($(md5sum "${NAME}/${SAMPLE_NAME}${MATE1_LABEL}.fastq.gz"))
+
+          zcat "${NAME}/${SAMPLE_NAME}${MATE1_LABEL}.fastq.gz" |
+          head -n 4000 |
+          awk '(NR + 2) % 4 == 0' > subsample.txt
+
+          LENGTH=`reads_length.py subsample.txt`
+
+          echo -e ""${SAMPLE_NAME}${MATE1_LABEL}.fastq.gz"\t${SAMPLE_NAME}\t${MD5}\t${PAIRED}\t${LENGTH}" >> "${NAME}".txt
+        {% endfor %}
+        {% if read|type|subtype('data:reads:fastq:paired:') %}
+          {% for fastq in read.fastq2 %}
+            cp {{ fastq.file }} "${NAME}/${SAMPLE_NAME}${MATE2_LABEL}.fastq.gz"
+            MD5=($(md5sum "${NAME}/${SAMPLE_NAME}${MATE2_LABEL}.fastq.gz"))
+
+            zcat "${NAME}/${SAMPLE_NAME}${MATE2_LABEL}.fastq.gz" |
+            head -n 4000 |
+            awk '(NR + 2) % 4 == 0' > subsample.txt
+
+            LENGTH=`reads_length.py subsample.txt`
+
+            echo -e ""${SAMPLE_NAME}${MATE2_LABEL}.fastq.gz"\t${SAMPLE_NAME}\t${MD5}\t${PAIRED}\t${LENGTH}" >> "${NAME}".txt
+          {% endfor %}
+        {% endif %}
+      {% endfor %}
+
+      {% for expression in expressions %}
+        SAMPLE_NAME=`basename {{ expression|sample_name }} .fastq.gz`
+
+        cp {{expression.exp.file}} "${NAME}"/"${SAMPLE_NAME}".tab.gz
+        MD5=($(md5sum "${NAME}/${SAMPLE_NAME}.tab.gz"))
+        echo -e ""${SAMPLE_NAME}".tab.gz\t${SAMPLE_NAME}\t${MD5}\t${SAMPLE_TO_PAIRED[${SAMPLE_NAME}]}" >> "${NAME}".txt
+      {% endfor %}
+
+      tar -zcvf "${NAME}".tar.gz "${NAME}"
+
+      EXP={{ expressions[0].exp_type }}
```

Comment:
`(expressions|first).exp_type`. But we may want to check that all expression data objects in the `expressions` list have the same expression type, source, species, build, and feature type.